include/aspect/utilities.h

```diff
@@ -813,6 +813,32 @@ namespace aspect
          */
         std_cxx11::unique_ptr<aspect::Utilities::AsciiDataLookup<1> > lookup;
     };
+
+    /**
+     * A weighted p norm average caluclator:
+     * $\text{weighed\_p\_norm\_average} =
+     * \frac{1}{k}(\frac{1}{k}(\sum_{i=1}^k \big(\text{weights}_i \text{values}_i^p)\big)^{\frac{1}{p}}\big)^{\frac{1}{p}}$
```

Comment:
I think you have the `^{1/p}` twice in the formula, right?

Also, can you name the arguments the same way you name them in the formula? I think the names in the formula make more sense.